datafusion/functions-aggregate/src/min_max.rs

```diff
@@ -113,8 +114,12 @@ macro_rules! primitive_max_accumulator {
     ($DATA_TYPE:ident, $NATIVE:ident, $PRIMTYPE:ident) => {{
         Ok(Box::new(
             PrimitiveGroupsAccumulator::<$PRIMTYPE, _>::new($DATA_TYPE, |cur, new| {
-                if *cur < new {
-                    *cur = new
+                match (new).partial_cmp(cur) {
+                    None | Some(Ordering::Greater) => {
```

Comment:
@alamb do you think, changing from `None | Some(Ordering::Greater)` to `Some(Ordering::Greater) | None` will make any difference because of short circuiting. If you think so I can change this and other comparison so that they benefit from short circuiting.

Reply:
I am not sure to be honest -- it wouldn't hurt to try it (I can also rerun the benchmarks too if you want)

Reply:
I just swapped the operations to be on the safe side, I don't think we need to re-run benchmarks (won't be visible probably even if makes change).